camayoc/tests/remote/quipucords/test_scan.py

```diff
@@ -0,0 +1,341 @@
+    and one scan will be run against the three sources each created with
+    their own credential.
+    """
+    config = get_config()
+    creds = config['credentials']
+    scans = config.get('qcs', {}).get('scans', [])
+    if scans == []:
+        # if no scans are defined, no need to go any further
+        return
+    inventory = {
+        machine['hostname']: machine for machine in config['inventory']
+    }
+    vcenter_inventory = {
+        machine['hostname']: machine for machine in config['inventory']
+        if machine.get('hypervisor') == 'vcenter'
+    }
+    if not creds or not inventory:
+        raise ValueError(
+            'Make sure to have credentials and inventory'
+            ' items in the config file'
+        )
+    vcenter_hostnames = list(vcenter_inventory.keys())
+    host_folder = vcenter_client.content.rootFolder \
+        .childEntity[VCENTER_DATA_CENTER].hostFolder
+    host = host_folder.childEntity[VCENTER_CLUSTER].host[VCENTER_HOST]
+    cred_ids = {}
+    source_ids = {}
+    for cred in creds:
+        # create creds on server
+        # create dict that associates the name of the cred to the cred id
+        cred_ids[cred['name']] = create_cred(cred, cleanup)
+    for hostname, source in inventory.items():
+        # create sources on server, and keep track of source ids for each scan
+        # name of cred to cred id on server
+        source_ids[hostname] = create_source(source, cred_ids, cleanup)
+        for scan in scans:
+            scan['source_ids'] = scan.get('source_ids', [])
+            for source in scan['sources']:
+                if hostname == source:
+                    scan['source_ids'].append(source_ids[hostname])
+    for scan in scans:
+        # update the sources dict of each item in the scans list with the
+        # source id if hosts are marked as being hosted on vcenter, turn them
+        # on. then wait until they are live
+        # then run the scan, caching the report id associated with the scan
+        # and finally turn the managed machines off.
+        vcenter_vms = [
+            vm for vm in host.vm
+            if (vm.name in vcenter_hostnames) and (vm.name in scan['sources'])
+        ]
+        machines_to_wait = []
+        for vm in vcenter_vms:
+            if vm.runtime.powerState == 'poweredOff':
+                vm.PowerOnVM_Task()
+                machines_to_wait.append(inventory[vm.name])
+        wait_until_live(machines_to_wait)
+        # now all host should be live and we can run the scan
+        # all results will be saved in global cache
+        # if errors occur, they will be saved but scanning will go on
+        run_scan(scan['source_ids'], scan['name'])
+        for vm in vcenter_vms:
+            if vm.runtime.powerState == 'poweredOn':
+                vm.PowerOffVM_Task()
+
+
+def scan_info():
+    """Generate list of scan dict objects found in config file."""
+    try:
+        cfg = get_config()
```

Comment:
This could be an oneliner: `return get_config().get('qcs', {}).get('scans', [])`